modules/services/gpg-agent.nix

```diff
@@ -352,8 +357,10 @@ in {
         launchd.agents.gpg-agent = {
           enable = true;
           config = {
-            ProgramArguments = [ "${gpgPkg}/bin/gpg-agent" "--supervised" ]
-              ++ optional cfg.verbose "--verbose";
+            # macOS doesn't like the "--supervised" option
+            ProgramArguments =
+              [ "${gpgPkg}/bin/gpgconf" "--launch" "gpg-agent" ]
+              ++ optionals cfg.verbose [ "--verbose" ];
```

Comment:
nice, thanks for updating this - i'm not sure if we'll need to look at changing how launchd manages/expects the lifecycle of the process to behave, given that `gpgconf --launch` basically schedules a separate process, then exits. so, perhaps `ProcessType = "Background"` isn't applicable here